### PR TITLE
docs: Fix up section headers in README

### PR DESCRIPTION
**Pull request by pmatulis**
* Repair table of contents - links were broken
* Remove documentation content (**Quickstart** section)
* Add **Documentation** section, and point to published content
* Replace **How Can I Contribute** section with **Project & community** section
* Add **License** section

Proposed organisation is based on consistency across projects. Some customisation is fine (e.g. table of contents).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -13,11 +13,11 @@
 
 
 ## Table of Contents
-1. [💡 Philosophy](#💡-philosophy)
-2. [🎯 Features](#🎯-features)
-3. [📖 Documentation](#📖-documentation)
-4. [⚡️ Quickstart](#⚡️-quickstart)
```
Comment (sabaini) [inline README.md:19]: I'm unsure about removing the "Quickstart" entirely. Although there certainly is some redundancy with the RTD tutorials, many developers will find the GH README first, and the quickstart could serve as showcase of how easy it is to get going with MicroCeph. We possibly should update and trim the quickstart though.

File: README.md [hunk 2]
```diff
@@ -33,118 +33,33 @@ Deploying and operating a Ceph cluster is complex because Ceph is designed to be
...
-## 👍 How Can I Contribute ?
-
-1. Checkout [MicroCeph Hacking Guide](/HACKING.md) to start building and contributing to the codebase.
-2. Excited about [MicroCeph](https://snapcraft.io/microceph) ? Join our [Stargazers](https://github.com/canonical/microceph/stargazers)
-3. Write reviews or tutorials to help spread the knowledge 📖
-4. Participate in [Pull Requests](https://github.com/canonical/microceph/pulls) and help fix [Issues](https://github.com/canonical/microceph/issues)
-
-You can also find us on Matrix @[Ubuntu Ceph](https://matrix.to/#/#ubuntu-ceph:matrix.org)
+The documentation is found in the [`docs`][docs-dir-microceph] directory. It is
+written in RST format, built with Sphinx, and published on Read The Docs:
```
Comment (sabaini) [inline README.md:37]: The RST format and using Sphinx for docs seems like an implementation detail, do we need to bring it up?

File: README.md (hunk 1 above)
Comment (pmatulis) [reply README.md:19]: As discussed, we'll keep the quickstart by making use of the loop files method.

**Review by sabaini — APPROVED** [review]
lgtm, thanks!

**Pull request merged**